### ISSUE TITLE
File "/usr/lib/python3.9/zipfile.py", line 1324, in _RealGetContents     raise BadZipFile("File is not a zip file")

### ISSUE THREAD

**Issue by chris2div** [sgherbst/pyhigh#1]
Hello, 

I want to use your fantastic package but I have a problem :-(.

When I launch a pyhigh command, I have this error msg. 

__Downloading N36W119.hgt.zip
Traceback (most recent call last):
  File "/usr/local/bin/pyhigh", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/dist-packages/pyhigh/pyhigh.py", line 23, in main
    print(get_elevation(lat=lat, lon=lon))
  File "/usr/local/lib/python3.9/dist-packages/pyhigh/elevation.py", line 69, in get_elevation
    return get_elevation_batch([(lat, lon)])[0]
  File "/usr/local/lib/python3.9/dist-packages/pyhigh/elevation.py", line 59, in get_elevation_batch
    unzip(CACHE_DIR / zip_name)
  File "/usr/local/lib/python3.9/dist-packages/pyhigh/unzip.py", line 6, in unzip
    zip_ref = zipfile.ZipFile(path, 'r')
  File "/usr/lib/python3.9/zipfile.py", line 1257, in __init__
    self._RealGetContents()
  File "/usr/lib/python3.9/zipfile.py", line 1324, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file__ 

Do you have any idea please ? 
Thanks a lot.

**Comment by jgbreault** [comment]
I have the same issue.

**Comment by cfangwen-uofm** [comment]
Original USGS site has closed the download of these data. Try this: 
```
def get_url_for_zip(zip_name):
    return f'https://firmware.ardupilot.org/SRTM/North_America/{zip_name}'
```
Change this line in elevation.py in the package.

**Comment by yousabg** [comment]
Hi! It's very late but I just want to share it works if you do this as well:
def get_url_for_zip(zip_name):
    return f'https://srtm.kurviger.de/SRTM3/North_America/{zip_name}'

**Comment by Seekeer** [comment]
Guys I need to run this lib on device without Internet. Could you please tell me how can I create a package with pre-downloaded archives? 
@yousabg @cfangwen-uofm

**Comment by yousabg** [comment]
If you already know which zip files you need you can predownload them and just put it in the same directory as your python file, if not you would have to download all of them which would be a lot

**Comment by CChapmanbird** [comment]
> USGS site has closed the download of these elevation data. Try this:
> 
> ```
> def get_url_for_zip(zip_name):
>     return f'https://firmware.ardupilot.org/SRTM/North_America/{zip_name}'
> ```
> 
> Change to this line in elevation.py in the package.

This change is added in #3 along with a general change to include data for other continents.

**Comment by dstone42** [comment]
@CChapmanbird Is this something that is likely to be fixed soon? I was hoping to use this package in a big project we are working on, but I can't be editing the source code of packages to get them to work for this kind of project. It needs to be portable and run on different machines. I need to be able to give a new machine my dependencies and have it download them and run my code. Is this a package I could use for that, or is it not likely this will be working soon enough and kept up to date?

**Comment by sgherbst** [comment]
@dstone42 and others - sorry for the epic delay!

I just released a new version, `0.0.6`, that has the download link fixed.  It's on PyPI, so `pip install "pyhigh>=0.0.6"` will make sure that you get a version that works.
